presto-native-sidecar-plugin/src/test/java/com/facebook/presto/sidecar/NativeSidecarPluginQueryRunnerUtils.java

```diff
@@ -30,12 +30,23 @@ public static void setupNativeSidecarPlugin(QueryRunner queryRunner)
         queryRunner.loadSessionPropertyProvider(
                 NativeSystemSessionPropertyProviderFactory.NAME,
                 ImmutableMap.of());
+
+        // Register native catalog for built-in functions
         queryRunner.loadFunctionNamespaceManager(
                 NativeFunctionNamespaceManagerFactory.NAME,
                 "native",
                 ImmutableMap.of(
                         "supported-function-languages", "CPP",
                         "function-implementation-type", "CPP"));
+
+        // Register hive catalog for hive-specific functions
+        queryRunner.loadFunctionNamespaceManager(
+                NativeFunctionNamespaceManagerFactory.NAME,
+                "hive",
```

Comment:
This assumes the PrestoServer has hive connector. Thats a reasonable assumption as we always setup hive connector. But then we should make the C++ PrestoServer code consistent by always registering the hive function as well (and not check if the hive connector is present)